scheduler/src/cook/kubernetes/metrics.clj

```diff
@@ -24,3 +27,28 @@
   "Given a metric name and a compute cluster name, returns a timer metric."
   [metric-name compute-cluster-name]
   (timers/timer (calculate-name metric-name compute-cluster-name)))
+
+(def histogram-supplier
+  (reify
+    MetricRegistry$MetricSupplier
+    (newMetric [_]
+      (Histogram.
+        ; The default implementation of `Reservoir` in dropwizard metrics is
+        ; `ExponentiallyDecayingReservoir`, which stores data samples for some
+        ; time. When new samples stop arriving, it uses the historical data and
+        ; returns the same characteristics for the data distribution again and
+        ; again, simply because the data distribution doesn’t change. Here we
+        ; switch from the default `ExponentiallyDecayingReservoir` to a sliding
+        ; time window reservoir, which gives zeros when there is no data. See
+        ; https://engineering.salesforce.com/be-careful-with-reservoirs-708884018daf
+        ; for more information.
+        (SlidingTimeWindowArrayReservoir. 30 TimeUnit/SECONDS)))))
```

Comment:
Do we want this to be longer, like 5 minutes?

Reply:
Yep, updated